### ISSUE TITLE
Update Resource Problems

### ISSUE THREAD

**Issue by ivanleomk** [Holmusk/HolmuskDB#113]
Outstanding

- [x] Update Terms
- [x]  wire up Search User by ID Endpoint
- [x]  Find a way to update state object with termID of new term that has been added
- [x]  Update Name endpoint

There are three main problems with the creation of a new term

1. I am unsure if the database allows us to check if the term has already been used for another user, allowing for a connection being made to that term.
2.  I do not have a way to update the termId of this new term into my state object. Currently the API post requests are made in the effects/updateResource.js reducer. This works fine for all of the post requests except the updating of terms. This is because the new term will be added to the database with a new term and the separation of concerns by using two reducers, one for state and one for API requests makes it difficult to update the termID of the new object since it is return asychronously. #114 
3. The current endpoint requires the provision of relations and labels with a new term. This is a problem because we only capture the name of the new term at the time of user input

**Comment by kwannoel** [comment]
@ivanleomk please update the endpoint you are referring to in `3.`

**Comment by kwannoel** [comment]
1. Our interface does not handle multi relationship queries
  ```
  person1-[:KNOWS]->person2->[:KNOWS]->person3
  ```

  It is safe to ignore this for now. It is tracked in https://github.com/Holmusk/HolmuskDB/issues/92

2. You can use a promise in this context.

In pseudo js:
```js
  query.post(<update term endpoint>)
           .then(response => accessTheTermId(response)) // Get the term id from the response
           .then(termId => dispatch(<ACTION_TYPE>, { termId }) // Update component state here using the reducer
           .catch(...)
```

3. Tracked here https://github.com/Holmusk/HolmuskDB/issues/116 and here #117

**Comment by kwannoel** [comment]
Thinking about point 3 more:
If we add `HAS` with every term we add, it is a rather hacky solution, as it is added in implicitly.

It is an issue if we `delete` relations as well. Deleting the last relation between a `person` and a `term` separates the 2 (essentially deleting the `term` for the `person`)

These can be circumvented by providing warnings for empty relation cells.
We can send a prompt along the lines of `"No relation to person"`

**Comment by kwannoel** [comment]
1. Tracked in #92 
2. Closed by #112 
3. Closed by #121
